CDS/js/test/youtube/TestYoutubeAuth.js

```diff
@@ -0,0 +1,230 @@
+        });
+
+        after(() => {
+            fsStub.restore();
+        });
+
+        beforeEach(() => {
+            stringSubstituteStub = sinon.stub(dataStore, 'substituteString', (connection, val) => {
+                return new Promise((fulfill) => {
+                    fulfill(val);
+                });
+            });
+        });
+
+        afterEach(() => {
+            stringSubstituteStub.restore();
+        });
+
+        it('should raise an exception if no path to credentials file provided', () => {
+
+            return assert.isRejected(youtubeAuth.getCredentials(), 'Cannot upload to youtube: client secrets file path not provided');
+        });
+
+        it('should fetch credentials when path is provided', () => {
+
+            process.env.client_secrets = CREDENTIALS_PATH;
+
+            assert.eventually.deepEqual(youtubeAuth.getCredentials(), { credentials: 'credentials'});
+            return;
+        });
+    });
+
+    describe('getAuthClient', () => {
+
+        var p12Stub;
+
+        beforeEach(() => {
+            p12Stub = sinon.stub(youtubeAuth, 'readP12');
+        });
+
+        afterEach(() => {
+            p12Stub.restore();
+        });
+
+        it('should raise an error if client_id is missing', () => {
+            var credentialsStub = sinon.stub(youtubeAuth, 'getCredentials');
+            var promise = new Promise((fulfill, reject) => {
+                fulfill({
+                    client_secret: 'secret',
+                    web: {client_email: 'client_email' }
+                });
+            });
+
+            credentialsStub.returns(promise);
+
+
+            assert.isRejected(youtubeAuth.getAuthClient(), 'Credentials file is missing client_id property');
+
+            credentialsStub.restore();
+            return;
+
+        });
+
+        it('should raise an error if client_secret is missing', () => {
+            var credentialsStub = sinon.stub(youtubeAuth, 'getCredentials');
+            var promise = new Promise((fulfill, reject) => {
+                fulfill({
+                    client_id: 'id',
+                    web: {client_email: 'client_email' }
+                });
+            });
+
+            credentialsStub.returns(promise);
+
+            assert.isRejected(youtubeAuth.getAuthClient(), 'Credentials file is missing a client_secret property');
+            credentialsStub.restore();
+            return;
+
+        });
+
+        it('should raise an error if web property is missing', () => {
+            var credentialsStub = sinon.stub(youtubeAuth, 'getCredentials');
+            var promise = new Promise((fulfill, reject) => {
+                fulfill({
+                    client_id: 'id',
+                    client_secret: 'secret',
+                });
+            });
+
+            credentialsStub.returns(promise);
+
+            assert.isRejected(youtubeAuth.getAuthClient(), 'Credentials file is missing client_email property');
```

Comment:
should this read 'Credentials file is missing web property' to match the test name?